packages/nextlove/src/types/index.ts

```diff
@@ -1,9 +1,15 @@
 import { NextApiResponse, NextApiRequest } from "next"
 import { Middleware as WrapperMiddleware } from "nextjs-middleware-wrappers"
+import { HttpException } from "nextjs-exception-middleware"
 import { z } from "zod"
 import { HTTPMethods } from "../with-route-spec/middlewares/with-methods"
 import { SecuritySchemeObject, SecurityRequirementObject } from "openapi3-ts"
 
+/** Export an interface version of HttpException to avoid class inheritance issues */
+export interface IHttpException
```

Comment:
Do we want to name this `IHttpException` to keep the naming consistent with the base class, or `IHttpError`?

I also tried [the suggested example from the github issue](https://github.com/seamapi/seam-connect/issues/2439) and it doesn't seem like `KnownDeviceError` can extend the interface either as it is right now:
<img width="945" alt="image" src="https://user-images.githubusercontent.com/15915700/227574235-6e873a2d-ea91-4b9b-9e2e-dc6af57dabe5.png">

Is the intention to consolidate the two types by adding `status`, `metadata`, `options` to `KnownDeviceError`?

Reply:
yep I believe so

class-based usage is discouraged so I'd say `IHttpError` is better to stay consistent with the error interfaces on Seam Connect, or maybe even just `HttpError`

Reply:
@seveibar @codetheweb just to confirm, is this what we're thinking for the IHttpError interface?
![image](https://user-images.githubusercontent.com/15915700/227654955-5a50302f-2e9e-44a9-afc8-6912963ff5c1.png)

We might have to add some attributes to `seam-connect` entities to implement this interface as shown in the screenshot.

For comparison, this is the existing `HttpException` class in nextlove:
```HttpException extends Error {
    status: number;
    metadata: HttpExceptionMetadata;
    options: ThrowingOptions;
    constructor(status: number, metadata: HttpExceptionMetadata, options?: ThrowingOptions);
    toString(): string;
}```

I need to take off for an appointment soon, but if we can get consensus on an approach I will merge later!

Reply:
yeah, I think we need `metadata` as well but otherwise looks good